pkg/gadgets/utils/test/helpers.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/cilium/ebpf"
+	"github.com/google/go-cmp/cmp"
+)
+
+// CreateMntNsFilterMap creates and fills an eBPF map that can be used
+// to filter by mount namespace in the different tracers.
+func CreateMntNsFilterMap(mountnsids ...uint64) (*ebpf.Map, error) {
+	const one = uint32(1)
+
+	mntnsSpec := &ebpf.MapSpec{
+		Name:       "my_filter_map",
+		Type:       ebpf.Hash,
+		KeySize:    8,
+		ValueSize:  4,
+		MaxEntries: 1024,
+	}
+	m, err := ebpf.NewMap(mntnsSpec)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, mountnsid := range mountnsids {
+		if err := m.Put(mountnsid, one); err != nil {
+			m.Close()
+			return nil, err
+		}
+	}
+
+	return m, nil
+}
+
+type Tracer interface {
+	Stop()
+}
+
+type Test[Event any] struct {
+	GetTracer     func(*Info, func(event Event)) Tracer
+	RunnerConfig  *Config
+	GenerateEvent func() error
+	// One of the returned events should match this
+	GetAnyExpectedEvent func(*Info) *Event
+	// Only one event should be returned and should match this
+	GetExpectedEvent func(*Info) *Event
+}
+
+type testRunner[Event any] struct {
+	Fatalf func(format string, args ...any)
+	t      *testing.T
+	tracer Tracer
+	runner *Runner
+}
+
+func NewTestRunner[Event any](t *testing.T) *testRunner[Event] {
+	testRunner := &testRunner[Event]{
+		t: t,
+	}
+
+	testRunner.Fatalf = func(format string, args ...any) {
+		if testRunner.runner != nil {
+			testRunner.runner.Close()
+		}
+
+		if testRunner.tracer != nil {
+			testRunner.tracer.Stop()
+		}
+
+		testRunner.t.Fatalf(format, args...)
+	}
+
+	return testRunner
+}
+
+func (t *testRunner[Event]) Run(tests map[string]Test[Event]) {
+	var err error
+
+	for name, test := range tests {
+		events := []Event{}
+		eventCallback := func(event Event) {
+			events = append(events, event)
+		}
+
+		t.runner, err = NewRunner(test.RunnerConfig)
+		if err != nil {
+			t.Fatalf("%q: failed to create runner: %s", name, err)
+		}
```

Comment:
How about running each test with `t.Run` so you don't have to handle `name` explicitly? And to allow running only some tests?

Reply:
ohman! This suggestion is what I was looking for, it improved the code a lot:
- testRunner is not needed anymore (it was some sort of magic structure to save variables and handle cleanup)
- if a test case fails other are executed (before the first call to Fatalf() would stop the execution of other test cases within the same test). 

I pushed a version with these changes (I did not amended the commit on purpose). Would you mind checking this up?

Reply:
Cool, I'll have a look :)

Reply:
OK, I was missing your newest comments. I realized keeping this without squash the commits was too hard, so I squashed the commits and pushed the new version.

Reply:
Looks good now, marking as resolved.